crates/rpc/rpc-eth-types/src/receipt.rs

```diff
@@ -65,29 +52,53 @@ where
 
 /// Converter for Ethereum receipts.
 #[derive(Debug)]
-pub struct EthReceiptConverter<ChainSpec> {
+pub struct EthReceiptConverter<
+    ChainSpec,
+    Builder = fn(Receipt, usize, TransactionMeta) -> ReceiptEnvelope<Log>,
+> {
     chain_spec: Arc<ChainSpec>,
+    build_rpc_receipt: Builder,
 }
 
-impl<ChainSpec> Clone for EthReceiptConverter<ChainSpec> {
+impl<ChainSpec, Builder> Clone for EthReceiptConverter<ChainSpec, Builder>
+where
+    Builder: Clone,
+{
     fn clone(&self) -> Self {
-        Self { chain_spec: self.chain_spec.clone() }
+        Self {
+            chain_spec: self.chain_spec.clone(),
+            build_rpc_receipt: self.build_rpc_receipt.clone(),
+        }
     }
 }
 
 impl<ChainSpec> EthReceiptConverter<ChainSpec> {
     /// Creates a new converter with the given chain spec.
     pub const fn new(chain_spec: Arc<ChainSpec>) -> Self {
-        Self { chain_spec }
+        Self {
+            chain_spec,
+            build_rpc_receipt: |receipt, next_log_index, meta| {
+                receipt.into_rpc(next_log_index, meta).into()
+            },
+        }
+    }
+
+    /// Sets new builder for the converter.
+    pub fn with_builder<Builder>(
+        self,
+        build_rpc_receipt: Builder,
+    ) -> EthReceiptConverter<ChainSpec, Builder> {
+        EthReceiptConverter { chain_spec: self.chain_spec, build_rpc_receipt }
     }
 }
 
-impl<N, ChainSpec> ReceiptConverter<N> for EthReceiptConverter<ChainSpec>
+impl<N, ChainSpec, Builder, Rpc> ReceiptConverter<N> for EthReceiptConverter<ChainSpec, Builder>
 where
-    N: NodePrimitives<Receipt = Receipt>,
+    N: NodePrimitives,
     ChainSpec: EthChainSpec + 'static,
+    Builder: Debug + Fn(N::Receipt, usize, TransactionMeta) -> Rpc + 'static,
```

Comment:
do we need to use this type as named type anywhere?

doesnt look like it so this closure seems fine

otherwise we probably need a helper for this